### PR TITLE
hal: infineon: Add Infineon assets for CAT1/PSoC 6 SoC

### PR DESCRIPTION
**Pull request by npal-cy**
Add Infineon assets for CAT1/PSoC 6 SoC
This repo contains Infineon Assets (PDL, HAL, etc), which are used in implementation of Zephyr drivers for Infineon devices.

This PR is reflected changes according to Task#2 for [RFC#42883.](https://github.com/zephyrproject-rtos/zephyr/issues/42883)

### CODE REVIEW

**Comment by npal-cy** [comment]
Hi @nandojve , @carlescufi, @parthitce could you please review the changes.

**Comment by gmarull** [comment]
A couple of points from my side:

- I'd like to see a PR on the Zephyr side that shows how that HAL is integrated, e.g., a driver using it. In cases like RPi/GD32 the PR introducing platform support was the one introducing the HAL on the Zephyr manifest.
- Why do we need stuff like https://github.com/Infineon/abstraction-rtos being part of the HAL? As far as I understand, the HAL should just provide APIs/helpers for hardware enablement. You can always create separate modules your customers can add to their applications if they need such functionality.

**Comment by npal-cy** [comment]
> I'd like to see a PR on the Zephyr side that shows how that HAL is integrated, e.g., a driver using it. In cases like RPi/GD32 the PR introducing platform support was the one introducing the HAL on the Zephyr manifest.

i will prepare something like  https://github.com/zephyrproject-rtos/zephyr/pull/42725 but a little bit smaller, with following  parts:
* Infineon PSoC 6 SOC Devicetree (only for one mpn to decrease number of file for review)
* Infineon CAT1/PSoC 6 SOC integration (only for one series to decrease number of file for review)
* Infineon CAT1 UART driver / binding for Infineon CAT1 UART driver
* Infineon CAT1 GPIO driver / binding for Infineon CAT1 GPIO driver
* Infineon CAT1 Pinctrl driver / binding for Infineon CAT1 Pinctrl driver
The cy8cproto_062_4343w board support, will out of score for this MR, we still working on Clock_control driver, which give us possibility to remove [bsp part](https://github.com/npal-cy/zephyr-1/tree/topic/infineon_cat1_psoc6_integration/boards/arm/cy8cproto_062_4343w/bsp) from board as was discussed in https://github.com/zephyrproject-rtos/zephyr/pulls?q=is%3Apr+is%3Aclosed+draft#discussion_r805623025

> Why do we need stuff like https://github.com/Infineon/abstraction-rtos being part of the HAL? As far as I understand, the HAL should just provide APIs/helpers for hardware enablement. You can always create separate modules your customers can add to their applications if they need such functionality.

Finally, we expect to enable cy8cproto_062_4343w board where we add Wifi and BT features. For Wifi, we are developing Zephyr wifi driver, which will use following Assets:
- [whd-bsp-integration](https://github.com/Infineon/whd-bsp-integration) and
- [wifi-host-driver](https://github.com/Infineon/wifi-host-driver)
both of them use [abstraction-rtos](https://github.com/Infineon/abstraction-rtos) as hal for rtos. Currently abstraction-rtos supports freertos, rtx, theadx, but we are working to add Zephyr as well, to avoid overhead with using compatible cmsis_v2.

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	url = https://github.com/Infineon/core-lib.git
+[submodule "psoc6cm0p"]
+	path = psoc6cm0p
+	url = https://github.com/Infineon/psoc6cm0p.git
+[submodule "abstraction-rtos"]
+	path = abstraction-rtos
+	url = https://github.com/Infineon/abstraction-rtos.git
+[submodule "whd-bsp-integration"]
+	path = whd-bsp-integration
+	url = https://github.com/Infineon/whd-bsp-integration.git
```
Comment (tejlmand) [inline .gitmodules:18]: why are you adding submodules here ?

Are you planning to fetch those through the west manifest ?
Cause if that is the case, then please read this section:
https://docs.zephyrproject.org/latest/guides/modules.html#module-repositories
> Module Repositories
>
>  - All modules included in the default manifest shall be hosted in repositories under the zephyrproject-rtos GitHub organization.

Would say the direction proposed here should be discussed at dev-review / TSC.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,76 @@
...
+  zephyr_library_sources_ifdef(COMPONENT_ZEPHYR abstraction-rtos/source/COMPONENT_ZEPHYR/cyabs_rtos_zephyr.c)
+endif()
+
+if(CONFIG_SOC_FAMILY_CAT1A)
+  # Add mtb-hal-cat1
+  zephyr_include_directories(mtb-hal-cat1/include/)
+  zephyr_include_directories(mtb-hal-cat1/COMPONENT_PSOC6HAL/include/)
+  zephyr_include_directories(mtb-hal-cat1/COMPONENT_PSOC6HAL/COMPONENT_CAT1A/include/)
+
+  FILE(GLOB hal_sources mtb-hal-cat1/COMPONENT_PSOC6HAL/source/*.c)
```
Comment (tejlmand) [inline CMakeLists.txt:30]: please no globbing, globbing is discourage by CMake.
https://cmake.org/cmake/help/latest/command/file.html#glob

> **Note** We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. 

so this should never be used for libraries.
It is only used for samples / tests-samples.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,76 @@
...
+  target_sources(app PRIVATE ${hal_sources})
+
+  FILE(GLOB hal_sources mtb-hal-cat1/COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/pin_packages/*.c)
+  target_sources(app PRIVATE ${hal_sources})
```
Comment (tejlmand) [inline CMakeLists.txt:34]: why is this added to the `app` library ?

**Review by tejlmand — CHANGES_REQUESTED** [review]
Thank you for opening this.

I think the submodule approach must be discussed at dev-review / TSC.

**Comment by tejlmand** [comment]
Feel free to follow git submodule discussion here: https://discord.com/channels/720317445772017664/946730141818699826

**Comment by npal-cy** [comment]
updated per https://discord.com/channels/720317445772017664/946730141818699826 the repo was update:
1. All assets are copied to hal_infineon (submodules are not allowed)
2. all assets have own CMakeLists.txt 
new structure:
```
\hal_infineon
|- abstraction-rtos
  |- **** - all abstraction-rtos content
  |- CMakeLists.txt (cmake for abstraction-rtos)
|- core-lib
  |- **** - all core-lib content
  |- CMakeLists.txt (cmake for core-lib)
|- mtb-hal-cat1
  |- **** - all mtb-hal-cat1 content
  |- CMakeLists.txt (cmake for mtb-hal-cat1)
|- mtb-pdl-cat1
  |- **** - all mtb-pdl-cat1 content
  |- CMakeLists.txt (cmake for mtb-pdl-cat1)
|- whd-bsp-integration
  |- **** - all whd-bsp-integration content
  |- CMakeLists.txt (cmake for whd-bsp-integration)
|- CMakeLists.txt (entry cmake)
```

File: CMakeLists.txt (hunk 3 above)
Comment (npal-cy) [reply CMakeLists.txt]: updated

File: CMakeLists.txt (hunk 2 above)
Comment (npal-cy) [reply CMakeLists.txt]: updated

File: .gitmodules (hunk 1 above)
Comment (npal-cy) [reply .gitmodules]: updated, now we do not use submodule for hal_infineon.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Thanks for the update.                                                                              
                                                                                                    
One question I have that I think must be discussed with a wider audience or maybe even taken to the TSC is the EULA included.
                                                                                                    
I don't have an answer, but I think it's an important question to raise.                            
                                                                                      
----------
                                                                                                    
I made some comments on the CMake code.                                                             
If you want to make it easier for yourself to upmerge those repos, then it's possible to move all CMake code to the Zephyr repo.
                                                                                                    
And you may use same tree structure under a `modules/hal_infineon/` folder as is used in the module itself.
                                                                                                    
For inspiration you may take a look at `hal_nordic` repo, where all Zephyr specific CMake code is in Zephyr itself.
https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/hal_nordic/CMakeLists.txt 
https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/hal_nordic/nrf_802154/CMakeLists.txt           
https://github.com/zephyrproject-rtos/hal_nordic/blob/master/zephyr/module.yml                      
                                                                                                    
Additional readings on the `cmake-ext: True`:                                                       
https://docs.zephyrproject.org/latest/guides/modules.html#module-integration-files-external

**Comment by nandojve** [comment]
Hi @npal-cy ,

I still thinking that we are touching to much in this initial phase. It is up to you can decide what you want inside hal_infineon but people from project still need check what will be there and make sure it is compliance with project rules. If we could just focus in the essential part, initially, that will speed up the process.

For instance, is mtb-pdl-cat1 the newer version of PDL? Can this directory structure replace what is inside hal_cypress?

**Comment by npal-cy** [comment]
> Hi @npal-cy ,
> 
> I still thinking that we are touching to much in this initial phase. It is up to you can decide what you want inside hal_infineon but people from project still need check what will be there and make sure it is compliance with project rules. If we could just focus in the essential part, initially, that will speed up the process.
> 
> For instance, is mtb-pdl-cat1 the newer version of PDL? Can this directory structure replace what is inside hal_cypress?

**Comment by npal-cy** [comment]
Hi @nandojve
> For instance, is mtb-pdl-cat1 the newer version of PDL? Can this directory structure replace what is inside hal_cypress?

yes, mtb-pdl-cat1 is newer version of PDL, we will use it instead  hal_cypress.

The directory structure for mtb-pdl-cat1 and other assets are inherited from their original repos (e.g. https://github.com/Infineon/mtb-pdl-cat1). We do not want to change it.

Except mtb-pdl-cat1 i added assets required for CAT1 integration (in different commits). I am preparing updating old PSOC 6 integration PR and introduce  CAT1 SoC PR, and they will need updated hal_infineon.

File: abstraction-rtos/EULA [hunk 4]
```diff
@@ -0,0 +1,55 @@
+CYPRESS END USER LICENSE AGREEMENT
+
+PLEASE READ THIS END USER LICENSE AGREEMENT ("Agreement") CAREFULLY BEFORE DOWNLOADING, INSTALLING, COPYING, OR USING THIS SOFTWARE AND ACCOMPANYING DOCUMENTATION.  BY DOWNLOADING, INSTALLING, COPYING OR USING THE SOFTWARE, YOU ARE AGREEING TO BE BOUND BY THIS AGREEMENT.  IF YOU DO NOT AGREE TO ALL OF THE TERMS OF THIS AGREEMENT, PROMPTLY RETURN AND DO NOT USE THE SOFTWARE.  IF YOU HAVE PURCHASED THIS LICENSE TO THE SOFTWARE, YOUR RIGHT TO RETURN THE SOFTWARE EXPIRES 30 DAYS AFTER YOUR PURCHASE AND APPLIES ONLY TO THE ORIGINAL PURCHASER.
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply abstraction-rtos/EULA:3]: discussion here: https://discord.com/channels/720317445772017664/946730141818699826

**Comment by npal-cy** [comment]
hi @tejlmand, 
I fixed cmakes per your comments. for now i keep all updates in different commits, finally i will rebase/squash them.

File: abstraction-rtos/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+
+# Add abstraction-rtos
+zephyr_include_directories(include/)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply abstraction-rtos/CMakeLists.txt]: fixed

File: core-lib/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Add core-lib
+zephyr_include_directories(include/)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply core-lib/CMakeLists.txt]: fixed

File: whd-bsp-integration/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Add whd-bsp-integration sources
+zephyr_include_directories(/)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply whd-bsp-integration/CMakeLists.txt]: fixed

File: mtb-hal-cat1/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Add mtb-hal-cat1
+zephyr_include_directories(include/)
+zephyr_include_directories(COMPONENT_PSOC6HAL/include/)
+zephyr_include_directories(COMPONENT_PSOC6HAL/COMPONENT_CAT1A/include/)
+
+# packages sources
+zephyr_library_sources(COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/pin_packages/cyhal_psoc6_01_104_m_csp_ble.c)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply mtb-hal-cat1/CMakeLists.txt:11]: added zephyr_library_named(modules_hal_infineon) in top cmake
https://github.com/zephyrproject-rtos/hal_infineon/pull/2/commits/04de13a651ea40a7f211f7a74e52fe82a1381cc0

File: mtb-pdl-cat1/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Add mtb-pdl-cat1
+zephyr_compile_definitions(${CONFIG_SOC_PART_NUMBER})
+zephyr_include_directories(cmsis/include/)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply mtb-pdl-cat1/CMakeLists.txt]: removed
https://github.com/zephyrproject-rtos/hal_infineon/pull/2/commits/c07b7ae670c4c4d72a44e95afc542f867a2f9090

File: mtb-pdl-cat1/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+zephyr_library_sources(drivers/source/cy_sysclk.c)
+zephyr_library_sources(drivers/source/cy_sysanalog.c)
+zephyr_library_sources(drivers/source/cy_ipc_pipe.c)
+zephyr_library_sources(drivers/source/cy_ipc_sema.c)
+zephyr_library_sources(drivers/source/cy_ipc_drv.c)
+zephyr_library_sources(drivers/source/cy_prot.c)
+zephyr_library_sources(drivers/source/cy_trigmux.c)
+zephyr_library_sources(drivers/source/cy_flash.c)
+
+zephyr_sources_ifdef(CONFIG_SOC_FAMILY_PSOC6 drivers/source/cy_sysint.c)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply mtb-pdl-cat1/CMakeLists.txt]: done

File: whd-bsp-integration/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+zephyr_library_sources(cybsp_wifi.c)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply whd-bsp-integration/CMakeLists.txt:7]: added zephyr_library_named(modules_hal_infineon) in top cmake
https://github.com/zephyrproject-rtos/hal_infineon/commit/04de13a651ea40a7f211f7a74e52fe82a1381cc0

**Comment by npal-cy** [comment]
Hi @nandojve,
https://github.com/zephyrproject-rtos/zephyr/pull/43494 - updated old PSoC 6 implementation to use mtb-pdl-cat1 from hal_infineon.

**Comment by npal-cy** [comment]
Hi @tejlmand and @carlescufi 

regarding to EULA usage. I propose the following updates:
1. add information about EULA in readme for this repo.
2. add note about  EULA in existing board documentation (can be done in scope of https://github.com/zephyrproject-rtos/zephyr/pull/43494)

Is it make sense for you? or it does not resolve the problem..

**Comment by npal-cy** [comment]
EULA files removed per decision: https://discord.com/channels/720317445772017664/951513032150896700/951567432340873216

**Comment by npal-cy** [comment]
Hi @parthitce,
i added test log for xmc device, to confirm that nothing was broken in existing xmc integration.
[xmc45_relax_kit_spot_check.txt](https://github.com/zephyrproject-rtos/hal_infineon/files/8251026/xmc45_relax_kit_spot_check.txt)
.

**Comment by parthitce** [comment]
> Hi @parthitce, i added test log for xmc device, to confirm that nothing was broken in existing xmc integration. [xmc45_relax_kit_spot_check.txt](https://github.com/zephyrproject-rtos/hal_infineon/files/8251026/xmc45_relax_kit_spot_check.txt) .

Many thanks, I will do a test for XMC. Unfortunately I will have access to my relax kit only by next week for testing.

**Review by tejlmand — DISMISSED** [review]
Thanks for getting the EULA issue handled.

I have taken a closer look at the CMake code.

Out of curiosity, are there any plans on updating the Kconfig file inside Zephyr tree with new settings, now that there are several new files added with this PR ?

**Comment by tejlmand** [comment]
What is status on this ?

> > I'd like to see a PR on the Zephyr side that shows how that HAL is integrated, e.g., a driver using it. In cases like RPi/GD32 the PR introducing platform support was the one introducing the HAL on the Zephyr manifest.
>
> i will prepare something like https://github.com/zephyrproject-rtos/zephyr/pull/42725 but a little bit smaller, with following parts:

**Comment by npal-cy** [comment]
> What is status on this ?
> 
> > > I'd like to see a PR on the Zephyr side that shows how that HAL is integrated, e.g., a driver using it. In cases like RPi/GD32 the PR introducing platform support was the one introducing the HAL on the Zephyr manifest.
> > 
> > 
> > i will prepare something like [zephyrproject-rtos/zephyr#42725](https://github.com/zephyrproject-rtos/zephyr/pull/42725) but a little bit smaller, with following parts:

Hi tejlmand,
It is in progress. Right now i setup the test (Twister), and found some issue in implementation. Working on fixing them.
i expect to push MR in best case end of this week, or beginning of next.

**Comment by npal-cy** [comment]
> Out of curiosity, are there any plans on updating the Kconfig file inside Zephyr tree with new settings, now that there are several new files added with this PR ?

yes we will update/add new for gpio, uart, i2c, counter, pinctrl, new HCI extension driver and WIFI .

File: abstraction-rtos/CMakeLists.txt [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+# Add abstraction-rtos
+zephyr_include_directories(include)
+
+# Enable CMSIS_RTOS_V2 support
+zephyr_include_directories_ifdef(CONFIG_CMSIS_RTOS_V2 include/COMPONENT_RTX)
+zephyr_library_sources_ifdef(CONFIG_CMSIS_RTOS_V2 source/COMPONENT_RTX/cyabs_rtos_rtxv5.c)
+
+# Enable Zephyr support
+zephyr_include_directories_ifdef(COMPONENT_ZEPHYR include/COMPONENT_ZEPHYR)
+zephyr_library_sources_ifdef(COMPONENT_ZEPHYR source/COMPONENT_ZEPHYR/cyabs_rtos_zephyr.c)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply abstraction-rtos/CMakeLists.txt:13]: we are working on adding support for of ZEPHYR in our abstraction-rtos.  COMPONENT_ZEPHYR will be configured in cmake file of drivers, which need abstraction-rtos. For now it is wifi driver.

File: mtb-hal-cat1/CMakeLists.txt [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+# Add mtb-hal-cat1
+zephyr_include_directories(include)
+zephyr_include_directories(COMPONENT_PSOC6HAL/include)
+zephyr_include_directories(COMPONENT_PSOC6HAL/COMPONENT_CAT1A/include)
+
+# packages sources
+zephyr_library_sources(COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/pin_packages/cyhal_psoc6_01_104_m_csp_ble.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/pin_packages/cyhal_psoc6_01_104_m_csp_ble_usb.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/pin_packages/cyhal_psoc6_01_116_bga_ble.c)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply mtb-hal-cat1/CMakeLists.txt:13]: i am working on update this.

**Comment by npal-cy** [comment]
> > What is status on this ?
> > > > I'd like to see a PR on the Zephyr side that shows how that HAL is integrated, e.g., a driver using it. In cases like RPi/GD32 the PR introducing platform support was the one introducing the HAL on the Zephyr manifest.
> > > 
> > > 
> > > i will prepare something like [zephyrproject-rtos/zephyr#42725](https://github.com/zephyrproject-rtos/zephyr/pull/42725) but a little bit smaller, with following parts:
> 
> Hi tejlmand, It is in progress. Right now i setup the test (Twister), and found some issue in implementation. Working on fixing them. i expect to push MR in best case end of this week, or beginning of next.

Added PR https://github.com/zephyrproject-rtos/zephyr/pull/44211

File: mtb-hal-cat1/CMakeLists.txt [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+zephyr_sources_ifdef(CONFIG_UART_CAT1 COMPONENT_PSOC6HAL/source/cyhal_uart.c)
+zephyr_sources_ifdef(CONFIG_ADC_CAT1 COMPONENT_PSOC6HAL/source/cyhal_adc_sar.c)
+zephyr_sources_ifdef(CONFIG_I2C_CAT1 COMPONENT_PSOC6HAL/source/cyhal_i2c.c)
+zephyr_sources_ifdef(CONFIG_SPI_CAT1 COMPONENT_PSOC6HAL/source/cyhal_spi.c)
+zephyr_sources_ifdef(CONFIG_COUNTER_CAT1 COMPONENT_PSOC6HAL/source/cyhal_timer.c)
+zephyr_sources_ifdef(CONFIG_LPTIMER_CAT1 COMPONENT_PSOC6HAL/source/cyhal_lptimer.c)
+zephyr_sources_ifdef(CONFIG_PWM_CAT1 COMPONENT_PSOC6HAL/source/cyhal_pwm.c)
+zephyr_sources_ifdef(CONFIG_RTC_CAT1 COMPONENT_PSOC6HAL/source/cyhal_rtc.c)
+zephyr_sources_ifdef(CONFIG_TRNG_CAT1 COMPONENT_PSOC6HAL/source/cyhal_trng.c)
+zephyr_sources_ifdef(CONFIG_WTD_CAT1 COMPONENT_PSOC6HAL/source/cyhal_wdt.c)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply mtb-hal-cat1/CMakeLists.txt]: Fixed

File: mtb-hal-cat1/CMakeLists.txt (hunk 13 above)
[the remark this answers is not in the document]
Comment (npal-cy) [reply mtb-hal-cat1/CMakeLists.txt]: fixed. updated pdl and hal cmake

File: mtb-pdl-cat1/CMakeLists.txt [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+zephyr_library_sources(drivers/source/cy_sd_host.c)
+zephyr_library_sources(drivers/source/cy_sysanalog.c)
+zephyr_library_sources(drivers/source/cy_sysclk.c)
+zephyr_library_sources(drivers/source/cy_syslib.c)
+zephyr_library_sources(drivers/source/cy_syspm.c)
+zephyr_library_sources(drivers/source/cy_systick.c)
+zephyr_library_sources(drivers/source/cy_tcpwm_counter.c)
+zephyr_library_sources(drivers/source/cy_tcpwm_pwm.c)
+zephyr_library_sources(drivers/source/cy_trigmux.c)
+zephyr_library_sources(drivers/source/cy_wdt.c)
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply mtb-pdl-cat1/CMakeLists.txt]: updated.

**Comment by aniruddhanalgirkar-eaton** [comment]
Hi,
We have a requirement of Wi-Fi communication between a Host controller and other WLAN chip. While evaluating, I found the muRata 1DX which uses CYW4343W (under hood) offers this solution.
So, I am working on the Wi-Fi connection between these using Zephyr framework and the WHD public API on the Host controller. And have gone through the - RTOS porting, utilizing existing HAL, SDIO mapping and starting the LWIP to Zephyr NET stack. 
Here, while creating the Network interface using the API - NET_DEVICE_INIT(…..), can the *WHD be used as a Wi-Fi driver to create such interface ? (*WHD - WHD along with other components such as - WCM, wifi-mw-core, whd-bsp-integration, core-lib, abstraction RTOS, network stack, PAL)
Also, may I know, by when the overall integration is expected ? This feature is awaited and wants to speed-up the development activity. Thanks in advance for your suggestions & guidance!

**Comment by npal-cy** [comment]
Hi @aniruddhanalgirkar-eaton,

> Here, while creating the Network interface using the API - NET_DEVICE_INIT(…..), can the *WHD be used as a Wi-Fi driver to create such interface ?

yes. We are currently working to add support CYWxxxx devices (WIFI and Bluetooth)

>It seems what's being proposed under this PR could be helpful in achieving my end-goal as well. I am very interested to know by when the overall integration is expected and

~ end of this month 


> also if I can contribute to speed-up the integration ? Thanks in advance for your suggestions & guidance!

I'll add you to wifi driver PR, you can provide your feedbacks.

Regards,
Nazar

**Comment by npal-cy** [comment]
Hi @tejlmand and @carlescufi,
All comments are fixed, can we proceed with merge?

File: abstraction-rtos/CMakeLists.txt [hunk 16]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# Add abstraction-rtos
+zephyr_include_directories(include)
+
+# Enable Zephyr support
+zephyr_include_directories_ifdef(CONFIG_ABSTRACTION_RTOS_COMPONENT_ZEPHYR include/COMPONENT_ZEPHYR)
+zephyr_library_sources_ifdef(CONFIG_ABSTRACTION_RTOS_COMPONENT_ZEPHYR source/COMPONENT_ZEPHYR/cyabs_rtos_zephyr.c)
```
Comment (nandojve) [inline abstraction-rtos/CMakeLists.txt:9]: Where are the Zephyr side code?

File: core-lib/CMakeLists.txt [hunk 17]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Add core-lib
+zephyr_include_directories(include)
```
Comment (nandojve) [inline core-lib/CMakeLists.txt:6]: This PR should be the first because of dependencies.

**Review by nandojve — CHANGES_REQUESTED** [review]
`hal: infineon: Add whd-bsp-integration asset for CYW43xxx wifi driver` this commit should be dropped; It is not part of sequence.

File: mtb-hal-cat1/CMakeLists.txt [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+# Common part
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_clock.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_gpio.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_hwmgr.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_interconnect.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_sdhc.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_syspm.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_system.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_utils.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_utils_psoc.c)
```
Comment (nandojve) [inline mtb-hal-cat1/CMakeLists.txt:69]: This uses some Zephyr Kconfig to enable code at hal_infineon. Problem is,majority of it doesn't have a driver at Zephyr side and maintainers wont be aware of this.

I don't like this approach because any changes on Zephyr Kconfigs can result in broken platform.

I think a better way is follow how other vendors are doing, ex. [GD32](https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/hal_gigadevice/CMakeLists.txt). In this example vendor define `CONFIG_USE_<VENDOR>_<PERIPHERAL>` which includes all necessary dependencies. It is defined at `hal_<VENDOR>` and controlled at driver Kconfig. In case of serial driver, for instance, [Serial Kconfig.gd32](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/Kconfig.gd32)

File: mtb-pdl-cat1/CMakeLists.txt [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+zephyr_library_sources_ifdef(CONFIG_ADC drivers/source/cy_sar.c)
+zephyr_library_sources_ifdef(CONFIG_I2C drivers/source/cy_scb_i2c.c)
+zephyr_library_sources_ifdef(CONFIG_SPI drivers/source/cy_scb_spi.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER drivers/source/cy_tcpwm_counter.c)
+zephyr_library_sources_ifdef(CONFIG_PWM drivers/source/cy_tcpwm_pwm.c)
+zephyr_library_sources_ifdef(CONFIG_LPTIMER_INFINEON_CAT1 drivers/source/cy_mcwdt.c)
+zephyr_library_sources_ifdef(CONFIG_RTC_INFINEON_CAT1 drivers/source/cy_rtc.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_FAMILY_PSOC6 drivers/source/cy_sysint.c)
+
+if(CONFIG_TRNG_INFINEON_CAT1)
```
Comment (nandojve) [inline mtb-pdl-cat1/CMakeLists.txt:31]: Where are defined the Kconfig with INFINEON_CAT1?

**Review by nandojve — CHANGES_REQUESTED** [review]

File: abstraction-rtos/CMakeLists.txt (hunk 16 above)
Comment (npal-cy) [reply abstraction-rtos/CMakeLists.txt]: Removed abstraction-rtos and whd-bsp-integration. will be added in separate PR.

File: core-lib/CMakeLists.txt (hunk 17 above)
Comment (npal-cy) [reply core-lib/CMakeLists.txt]: updated

File: mtb-pdl-cat1/CMakeLists.txt (hunk 19 above)
Comment (npal-cy) [reply mtb-pdl-cat1/CMakeLists.txt]: https://github.com/zephyrproject-rtos/zephyr/pull/44211/files#diff-aac2c98a0043e7c1c60d01b3c48623fb861aa58ca8fa7a6bb2a47d0da9dfb91c (from PR#44211)

in scope of this PR i remove any CAT1 configs, will add to PR#44211.
now we have easy handle update Cakefiles in zephyr/modules/hal_infineon (https://github.com/zephyrproject-rtos/zephyr/pull/43494/commits/dd34a7e41a9fe0869412012b0347bef040da0c6b)

File: CMakeLists.txt [hunk 20]
```diff
@@ -1,6 +1,27 @@
-# SPDX-License-Identifier: Apache-2.0
-#
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
 # Copyright (c) 2020 Linumiz
-# Author: Parthiban Nallathambi <parthiban@linumiz.com>
+# SPDX-License-Identifier: Apache-2.0
```
[the remark this answers is not in the document]
Comment (npal-cy) [reply CMakeLists.txt]: updated

File: mtb-pdl-cat1/CMakeLists.txt (hunk 19 above)
Comment (nandojve) [reply mtb-pdl-cat1/CMakeLists.txt]: Ok. don't forget to adjust each commit properly.

**Review by nandojve — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by gmarull — CHANGES_REQUESTED** [review]
Why does this PR include things like `arm_math.h` (part of CMSIS-DSP), toolchain files (e.g. `TOOLCHAIN_IAR` folder), `*.cypersonality` files...? Is all that required to work with Zephyr? Why?

**Comment by npal-cy** [comment]
> Why does this PR include things like `arm_math.h` (part of CMSIS-DSP), toolchain files (e.g. `TOOLCHAIN_IAR` folder), `*.cypersonality` files...? Is all that required to work with Zephyr? Why?

Not all. 
mtb-pdl-cat1, mtb-hal-cat1, core-lib these are libraries, which we released (on Github) and uses in different environments like ModusToolbox, Mbed, AFR). 

It is our policy to use those libraries as is, so we can have perfect version tracking. we would normally use a git submodule to point at the official release, but Zephyr standards don't support that (https://discord.com/channels/720317445772017664/946730141818699826), so we had to copy it in.

So there are some part of content of those library which are not used by current integration. Some of them can be used in future when we will extend more Zephyr drivers (Crypro, spi). Some of them will never use (like *.cypersonality or CMSIS content from mtb-pdl-cat1\cmsis\include). The stuff which we used for Zephyr is described in Makefiles (\zephyr\modules\hal_infineon\**) 

I see another vendors also has similar (copy of assets as is):
 - modules\hal\nxp\mcux\mcux-sdk
 - modules\hal\silabs\gecko\Device\SiliconLabs\EFM32HG\Source\(IAR/GCC/ARM) support different toolchains. In our case (in PDL or other libraries) all code which are different for other toolchains, locates in specific folders (e.g TOOLCHAIN_IAR)

**Comment by gmarull** [comment]
@npal-cy thanks for the explanation, seems reasonable if you're taking released packages as they are.
